public/js/contribute/pad/save.js

```diff
@@ -88,88 +91,124 @@ function retrieveItems(kwargs) {
+        // text += `${datum.instruction}\n`;
+        text += `${sel.select('.instruction').node().innerText}\n`;
+      }
+      text += `${innerText}\n`;
+    }
+    return { item: datum, text };
   } else if (['checklist', 'radiolist'].includes(datum.type)) {
     datum.has_content =
       datum.options.filter((b) => b.name?.length && b.checked).length > 0;
     const clone = JSON.parse(JSON.stringify(datum));
     clone.options = clone.options.filter((b) => b.name?.length);
     // items.push(clone);
     // SET THE fullTxt REPRESENTATION
-    if (store_instructions && datum.instruction) {
-      // text += `${clone.instruction}\n`;
-      text += `${sel.select('.instruction').node().innerText}\n`
+    let innerText = '';
+    if (datum.has_content)
+      innerText += clone.options
+        .filter((b) => b.name?.length && b.checked)
+        .map((b) => b.name)
+        .join('\n');
+    innerText = innerText.trim();
+    if (innerText.length) {
+      if (store_instructions && datum.instruction) {
+        // text += `${clone.instruction}\n`;
+        text += `${sel.select('.instruction').node().innerText}\n`;
+      }
+      text += `${innerText}\n`;
     }
-    if (datum.has_content) text += clone.options.filter((b) => b.name?.length && b.checked).map((b) => b.name).join('\n');
 
     // datum.options = datum.options.filter(b => b.name && b.name.length)
     // items.push(datum)
-    return { item: clone, text }
+    return { item: clone, text };
   }
   // SPECIFIC META
   else if (datum.type === 'location') {
     datum.has_content = datum.centerpoints?.length > 0;
     // items.push(datum);
     // SET THE fullTxt REPRESENTATION
-    if (store_instructions && datum.instruction) {
+    if (store_instructions && datum.instruction && datum.has_content) {
       // text += `${datum.instruction}\n`;
-      text += `${sel.select('.instruction').node().innerText}\n`
+      text += `${sel.select('.instruction').node().innerText}\n`;
     }
     // NO SYSTEMATIC WAY OF GETTING location FOR fullTxt
-    return { item: datum, text }
-
+    return { item: datum, text };
   } else if (['tag', 'index'].includes(datum.type)) {
     datum.has_content = (datum.sdgs?.length || datum.tags?.length) > 0; // THIS IS LEGACY FOR THE ACTION PLANNING PLATFORM: TO DO: DEPRECATE
     // items.push(datum);
     // SET THE fullTxt REPRESENTATION
-    if (store_instructions && datum.instruction) {
-      // text += `${datum.instruction}\n`;
-      text += `${sel.select('.instruction').node().innerText}\n`
+    let innerText = '';
+    if (datum.has_content)
+      innerText += (datum.sdgs || datum.tags)
+        .map((b) => `${b.type}: ${b.name}`)
+        .join('\n');
+    innerText = innerText.trim();
+    if (innerText.length) {
+      if (store_instructions && datum.instruction) {
+        // text += `${datum.instruction}\n`;
+        text += `${sel.select('.instruction').node().innerText}\n`;
+      }
+      text += `${innerText}\n`;
```

Comment:
Same here: why not just use datum.has_content?

Reply:
having the arrays not empty doesn't guarantee that the text won't be empty. that said, do we want both sdgs and tags here? that seems better. right now we show the tags only if sdgs is empty